[Sim Justin] iP

1simjustin opened the pull request.

1simjustin committed:
Rename, add greet and exit

1simjustin committed:
Merge pull request #1 from 1simjustin/increment/level-0

Rename, add greet and exit

1simjustin committed:
Add function for logo and line string

1simjustin committed:
Add init and exit functions

1simjustin committed:
Add echo and exit

1simjustin committed:
Merge pull request #2 from 1simjustin/increment/level-1

Increment/level 1

1simjustin committed:
Add marker for user input

1simjustin committed:
Add List class

1simjustin committed:
Add add to list function

1simjustin committed:
Merge pull request #3 from 1simjustin/increment/level-2

Increment/level 2

1simjustin committed:
Completed increment level 3
+ Added Task class
+ Updated List class to use Task objects
+ Added mark and unmark functionalities for input

1simjustin committed:
Add error catching for task index

1simjustin committed:
Merge pull request #4 from 1simjustin/increment/level-2

Increment/level 3

1simjustin committed:
Add function for input parsing
+ Rewrote some bot lines

1simjustin committed:
Add Task types
+ Created Todo, Deadline, Event tasks using inheritance
+ Add keyword parsing to create new task types

1simjustin committed:
Add input function supporting leading string

1simjustin committed:
Add support for todo, deadline, event
+ No error handling

1simjustin committed:
Improve readability for addItem method

1simjustin committed:
Improve flavour text

1simjustin committed:
Merge pull request #5 from 1simjustin/increment/level-4

Increment/level 4

1simjustin committed:
Improve method names

1simjustin committed:
Shift input parsing into its own method

1simjustin committed:
Fix bug for loop exit

1simjustin committed:
Merge pull request #6 from 1simjustin/refactor/code-quality

Refactor/code quality

1simjustin committed:
Add error handling for invalid addItem

1simjustin committed:
Update error catching using try-catch

1simjustin committed:
Add text ui testing input and expected output

1simjustin committed:
Add edge case, empty command

1simjustin committed:
Merge pull request #7 from 1simjustin/addition/text-ui-testing

Addition/text-ui-testing

ZMinghuiZ commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,173 @@
+import java.util.Scanner;
+
 public class Duke {
```
it would be good if the class name can be changed to ur chatbot's name

ZMinghuiZ commented on src/main/java/List.java:
```diff
@@ -0,0 +1,163 @@
+
+        try {
+            switch (type) {
+            case "T":
```
Naming of the case can be more descriptive :)

ZMinghuiZ commented on src/main/java/List.java:
```diff
@@ -0,0 +1,163 @@
+import java.util.ArrayList;
+
+public class List {
```
Class name can be more descriptive since this class mainly is about task list

ZMinghuiZ commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+     * @return String representation of the task.
+     */
+    public String toString() {
+        return "[" + this.type + "]" + "[" + (this.isDone ? "X" : " ") + "] " + this.name
```
can use String.format to shorten the expression

ZMinghuiZ reviewed (commented):
Good job! nice documentation and exception control👍

JeffinsonDarmawan commented on src/main/java/List.java:
```diff
@@ -91,44 +107,57 @@ public void addItem(String taskName) {
      * @return None
      */
     public void addItem(String description, String type) {
```
Really love the fact that you have not written in an arrowhead style code, but this method looks very long. Perhaps this might pose as a challenge for when you are trying to debug this method.

1simjustin committed:
Refactor coding style
+ Change file name for List class
+ Use String.format method for toString output
+ Add TODO for enum to handle type in addItem method

1simjustin committed:
Add enum for parsing of task types

1simjustin committed:
Fix compile error

1simjustin committed:
Merge pull request #8 from 1simjustin/refactor/improve-coding-style

Refactor/improve-coding-style

1simjustin committed:
Change addItem call
+ Combine call across different task types
+ Update task type abbrievation to be based on TaskType enum

1simjustin committed:
Add more autotesting

1simjustin committed:
Merge pull request #9 from 1simjustin/refactor/improve-coding-style

Refactor/improve coding style

1simjustin committed:
Publish and rename

1simjustin committed:
Revert "Publish and rename"

This reverts commit 4431a6531eb8aaf2641be43fd89d3b0052de8678.

1simjustin committed:
Rename main

1simjustin committed:
Merge pull request #10 from 1simjustin/branch-Level-5

Publish and rename

1simjustin committed:
Place files into package

1simjustin committed:
Update directory for text ui testing

1simjustin committed:
Add exception class

1simjustin committed:
Add packages
+ Abstract Handler class for error handling methods
+ UserInput and Printer classes

1simjustin committed:
Place various segments into packages
+ Added class-specific exception handler UwunzheException
+ Improved exception handling using exception handler
+ Placed input handling into its own class

1simjustin committed:
Update autotesting input file for more extensive tests

1simjustin committed:
Bugfix autotesting script and updated EXPECTED

1simjustin committed:
Merge pull request #11 from 1simjustin/branch-A-Packages

Group classes into packages

1simjustin committed:
Add functionality for delete command
+ Change task list to static type
+ Shift printing of task with/without leading string to a function
+ Add printing of task based on index

1simjustin committed:
Merge pull request #12 from 1simjustin/branch-Level-6

Add functionality for delete command

1simjustin committed:
Add StorageHandler class and object instantiation

1simjustin committed:
Add load data from file
+ If file does not exist or file was empty, resulting task list will be empty as well

1simjustin committed:
Add save to file functionality
+ Update .gitignore to ignore data file
+ Add save to file when list is updated
+ Add method toStorageString for task types to convert to string for saving

1simjustin committed:
Merge pull request #13 from 1simjustin/branch-Level-7

Branch level 7

1simjustin committed:
Merge pull request #14 from 1simjustin/master

Update with level-6

1simjustin committed:
Change detailed below
+ Add init success verification
+ Add additional documentation
+ Update text ui test to test saving

1simjustin committed:
Merge pull request #15 from 1simjustin/branch-Level-7

Branch level 7

1simjustin committed:
Update file path for storage

1simjustin committed:
Merge pull request #16 from 1simjustin/addition/JAR

Update file path for storage

1simjustin committed:
Extract Ui, Storage, Parser, TaskList classes

1simjustin committed:
Extract Command classes to streamline main logic

1simjustin committed:
Remove unused package dependencies

1simjustin committed:
Merge pull request #17 from 1simjustin/addition/moreOOP

Addition/more oop

1simjustin committed:
Improve method procedure for setStatus and deleteItem

1simjustin committed:
Add find command class
+ Update error catching for invalid list and exit operations
+ Find command is not yet implemented

1simjustin committed:
Merge Printer class and Ui class to reduce overlap

1simjustin committed:
Implement Find function
+ Update text ui testing
+ Change printing functions to static

1simjustin committed:
Remove unused package dependencies

1simjustin committed:
Merge pull request #19 from 1simjustin/branch-Level-9

Improve method procedure for setStatus and deleteItem

1simjustin committed:
Merge pull request #21 from 1simjustin/master

Update branch lv 8

1simjustin committed:
Merge pull request #22 from 1simjustin/master

Update branch javadoc

wwweert123 commented on src/main/java/uwunzhe/commands/MarkUnmarkCommand.java:
```diff
@@ -0,0 +1,40 @@
+            storage.saveData(taskList);
+
+        } catch (IndexOutOfBoundsException e) {
+            throw new UwunzheException("Huhhhhhhh? I cannot find!");
```
Can define error message constants at the top of class to prevent use of magic literals

wwweert123 commented on src/main/java/uwunzhe/handler/Parser.java:
```diff
@@ -0,0 +1,65 @@
+        case "list":
+            return new ListCommand(command, taskString);
+
+        case "mark":
```
use a "//fallthrough" comment to indicate a case without break statement

wwweert123 reviewed (commented):
Amazing code, highly OOPed, almost no coding standard violations, very upz

1simjustin committed:
Update javadoc

1simjustin committed:
Update text ui testing

1simjustin committed:
Add enum for exception types

1simjustin committed:
Merge pull request #23 from 1simjustin/branch-A-JavaDoc

Branch a java doc

1simjustin committed:
Merge pull request #24 from 1simjustin/master

Update branch

1simjustin committed:
Add DateHandler class for conversion to LocalDate type
+ Add ignore invalid lines in storage data
+ Update relevant Task types to store as LocalDate type
+ Update relevant UI and storage outputs from LocalDate to String type
+ Pending update to text ui testing

1simjustin committed:
Update text ui testing
+ Add check for event date range

1simjustin committed:
Add date command

1simjustin committed:
Update text ui testing to include date command

1simjustin committed:
Merge pull request #25 from 1simjustin/branch-Level-8

Branch level 8

1simjustin committed:
Update user guide

1simjustin committed:
Test quick reference

1simjustin committed:
Test quick reference

1simjustin committed:
Test quick reference

1simjustin committed:
Update quick reference with links

1simjustin committed:
Update quick reference

1simjustin committed:
Update readme

1simjustin committed:
Add ToC

1simjustin committed:
Merge pull request #26 from 1simjustin/addition/user-guide

Addition/user guide
